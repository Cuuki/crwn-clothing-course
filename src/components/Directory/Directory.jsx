import React from 'react';

import {connect} from 'react-redux';
import {createStructuredSelector} from 'reselect';
import {selectDirectorySections} from '../../selectors/directorySelectors';

import DirectoryItem from '../DirectoryItem';

import './Directory.scss';

const Directory = ({sections}) => (
  <div className="directory">
    {sections.map(({id, ...otherProps}) => {
      return <DirectoryItem key={id} {...otherProps} />;
    })}
  </div>
);

const mapStateToProps = createStructuredSelector({
  sections: selectDirectorySections,
});

export default connect(mapStateToProps)(Directory);
