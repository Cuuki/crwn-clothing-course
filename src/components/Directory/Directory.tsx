import React from 'react';

import DirectoryItem from '../DirectoryItem';

import './Directory.scss';

const Directory = ({sections}) => (
  <div className="directory">
    {sections.map(({id, ...otherProps}) => {
      return <DirectoryItem key={id} {...otherProps} />;
    })}
  </div>
);

export default Directory;
