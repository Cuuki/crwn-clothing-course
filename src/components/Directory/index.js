import {connect} from 'react-redux';
import {createStructuredSelector} from 'reselect';
import {selectDirectorySections} from '../../redux/modules/directory';
import Directory from './Directory';

const mapStateToProps = createStructuredSelector({
  sections: selectDirectorySections,
});

export default connect(mapStateToProps)(Directory);
