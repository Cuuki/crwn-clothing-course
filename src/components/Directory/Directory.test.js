import React from 'react';
import {MemoryRouter} from 'react-router-dom';
import {render} from "@testing-library/react";
import Directory from './Directory';

describe('Directory', () => {
  it('renders Directory component', () => {
    const mockSections = [
      {
        title: 'test',
        imageUrl: 'https://i.ibb.co/cvpntL1/hats.png',
        id: 1,
        linkUrl: 'shop/test',
      },
      {
        title: 'test 2',
        imageUrl: 'https://i.ibb.co/cvpntL1/hats.png',
        id: 2,
        linkUrl: 'shop/test2',
      },
    ];

    expect(render(
      <Directory sections={mockSections}/>,
      {wrapper: MemoryRouter})
    ).toMatchSnapshot();
  });
})
