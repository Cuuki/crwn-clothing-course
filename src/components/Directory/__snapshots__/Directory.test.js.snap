// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directory renders Directory component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="directory"
      >
        <div
          class="directory-item"
        >
          <div
            class="background-image"
            style="background-image: url(https://i.ibb.co/cvpntL1/hats.png);"
          />
          <div
            class="content"
          >
            <h1
              class="title"
            >
              TEST
            </h1>
            <span
              class="subtitle"
            >
              SHOP NOW
            </span>
          </div>
        </div>
        <div
          class="directory-item"
        >
          <div
            class="background-image"
            style="background-image: url(https://i.ibb.co/cvpntL1/hats.png);"
          />
          <div
            class="content"
          >
            <h1
              class="title"
            >
              TEST 2
            </h1>
            <span
              class="subtitle"
            >
              SHOP NOW
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="directory"
    >
      <div
        class="directory-item"
      >
        <div
          class="background-image"
          style="background-image: url(https://i.ibb.co/cvpntL1/hats.png);"
        />
        <div
          class="content"
        >
          <h1
            class="title"
          >
            TEST
          </h1>
          <span
            class="subtitle"
          >
            SHOP NOW
          </span>
        </div>
      </div>
      <div
        class="directory-item"
      >
        <div
          class="background-image"
          style="background-image: url(https://i.ibb.co/cvpntL1/hats.png);"
        />
        <div
          class="content"
        >
          <h1
            class="title"
          >
            TEST 2
          </h1>
          <span
            class="subtitle"
          >
            SHOP NOW
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
